src/core/executeCommandFactory.js

```diff
@@ -94,6 +94,13 @@ export default ({
       const executor = getExecutor(commandName, options);
       logger.log(`Executing ${commandName} command.`, "Options:", options);
       resolve(executor());
-    }).catch(handleError);
+    })
+      .then(result => {
+        // We should always be returning an object from every command.
+        return isObject(result) ? result : {};
```

Comment:
This is important to call out. Any time a component's command does not return an object or a promise which resolves to an object, we make sure we return an object to the customer. An alternative to this is to make every component command return an object from within the component's code itself, but we run a greater risk of an engineering creating a new command and forgetting to return an object.

We always return a `result` object just to be consistent across commands.